scripts/cmake/vcpkg_backup_restore_env_vars.cmake

```
@@ -0,0 +1,55 @@
+#[===[.md:
+# vcpkg_backup_restore_env_vars
+
+Backup or restore the environment variables
+
+## Usage:
+```cmake
+vcpkg_backup_env_variables(VARS <ENV_VARS>)
+```
+
+```cmake
+vcpkg_restore_env_variables(VARS <ENV_VARS>)
+```
+
+### ENV_VARS
+The target passed to the make build command (`./make <target>`). If not specified, the 'all' target will
+be passed.
+
+## Notes:
+This command should be preceded by a call to [`vcpkg_configure_make()`](vcpkg_configure_make.md).
+You can use the alias [`vcpkg_install_make()`](vcpkg_install_make.md) function if your makefile supports the
+"install" target
+
+#]===]
+
+function(z_vcpkg_backup_env_variable envvar)
+    if(DEFINED ENV{${envvar}})
+        set(${envvar}_backup "$ENV{${envvar}}" PARENT_SCOPE)

```

probably best to use `z_vcpkg_env_backup_${envvar}` instead, since it's an internal variable
```suggestion
        set("z_vcpkg_env_backup_${envvar}" "$ENV{${envvar}}" PARENT_SCOPE)
```